scala-package/core/src/main/scala/org/apache/mxnet/IO.scala

```diff
@@ -228,15 +264,14 @@ object DataBatch {
 
     /**
      * Provide the shape of a data.
-     * @param name data name.
-     * @param shape data shape.
+     * @param dataDesc DataDescriptor
      * @return this.
      */
-    def provideDataShape(name: String, shape: Shape): Builder = {
-      if (datatShapes == null) {
-        datatShapes = ListMap((name, shape))
+    def provideDataShape(dataDesc: DataDesc): Builder = {
+      if (dataShapes == null) {
+        dataShapes = IndexedSeq(dataDesc)
       } else {
-        datatShapes = datatShapes.updated(name, shape)
+        dataShapes = dataShapes ++ IndexedSeq(dataDesc)
```

Comment:
provideDataShape->`provideDataDesc`
this does not look right. which Index of the Input do you want to add to?.  You should maintain the same semantic as in the constructor ie., take `IndexedSeq[DataDesc]` as arguments for Builder methods. 
If this is to keep it compatible with Java, I suggest you create another method that takes `IndexedSeq[DataDesc]`..although i think Scala might cause problems due to erasure

Reply:
Do you want to the same as you what did for NDArrayIter.builder

Reply:
@yzhliu the previous builder only allows single dataDesc (shape, name) input for the DataBatch. However, the data passed in is a IndexedSeq, in the new design, I changed them into indexedSeq and remove the old method. 

@nswamy About NDArrayIter, they are single input for both data and DataDesc, I am not planning to do the same thing there.